openstudiocore/src/model/GeneratorPVWatts.cpp

```diff
@@ -0,0 +1,443 @@
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::SystemLosses, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isSystemLossesDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::SystemLosses);
+  }
+
+  double GeneratorPVWatts_Impl::tiltAngle() const {
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::TiltAngle, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isTiltAngleDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::TiltAngle);
+  }
+
+  double GeneratorPVWatts_Impl::azimuthAngle() const {
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::AzimuthAngle, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isAzimuthAngleDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::AzimuthAngle);
+  }
+
+  boost::optional<PlanarSurface> GeneratorPVWatts_Impl::surface() const {
+    return getObject<ModelObject>().getModelObjectTarget<PlanarSurface>(OS_Generator_PVWattsFields::SurfaceName);
+  }
+
+  double GeneratorPVWatts_Impl::groundCoverageRatio() const {
+    boost::optional<double> value = getDouble(OS_Generator_PVWattsFields::GroundCoverageRatio, true);
+    OS_ASSERT(value);
+    return value.get();
+  }
+
+  bool GeneratorPVWatts_Impl::isGroundCoverageRatioDefaulted() const {
+    return isEmpty(OS_Generator_PVWattsFields::GroundCoverageRatio);
+  }
+
+  bool GeneratorPVWatts_Impl::setDCSystemCapacity(double dcSystemCapacity) {
+    bool result = setDouble(OS_Generator_PVWattsFields::DCSystemCapacity, dcSystemCapacity);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  bool GeneratorPVWatts_Impl::setModuleType(std::string moduleType) {
+    bool result = setString(OS_Generator_PVWattsFields::ModuleType, moduleType);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetModuleType() {
+    bool result = setString(OS_Generator_PVWattsFields::ModuleType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setArrayType(std::string arrayType) {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, arrayType);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetArrayType() {
+    bool result = setString(OS_Generator_PVWattsFields::ArrayType, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setSystemLosses(double systemLosses) {
+    bool result = setDouble(OS_Generator_PVWattsFields::SystemLosses, systemLosses);
+    OS_ASSERT(result);
+    return result;
+  }
+
+  void GeneratorPVWatts_Impl::resetSystemLosses() {
+    bool result = setString(OS_Generator_PVWattsFields::SystemLosses, "");
+    OS_ASSERT(result);
+  }
+
+  bool GeneratorPVWatts_Impl::setTiltAngle(double tiltAngle) {
+    bool result = setDouble(OS_Generator_PVWattsFields::TiltAngle, tiltAngle);
```

Comment:
Might also blank out surface?  If you don't clear the surface, then the translator would not use this value.

Reply:
Yeah I think that'd be ok. Azimuth angle too right?